staging/src/k8s.io/kubectl/pkg/cmd/auth/cani.go

```diff
@@ -416,6 +434,40 @@ func isNamespaced(gvr schema.GroupVersionResource, discoveryClient discovery.Dis
 	return false, fmt.Errorf("the server doesn't have a resource type '%s' in group '%s'", gvr.Resource, gvr.Group)
 }
 
+func hasSubresource(discoveryClient discovery.DiscoveryInterface, gvr schema.GroupVersionResource, subresource string) error {
+	groupVersion := gvr.GroupVersion().String()
+	resourceAndSubresource := fmt.Sprintf("%v/%v", gvr.Resource, subresource)
+
+	var serverResources []*metav1.APIResourceList
+	if len(groupVersion) > 0 {
+		var apiResourceList *metav1.APIResourceList
+		apiResourceList, _ = discoveryClient.ServerResourcesForGroupVersion(groupVersion)
+		if apiResourceList != nil {
+			serverResources = []*metav1.APIResourceList{apiResourceList}
+		}
+	}
+	if serverResources == nil {
+		_, serverResources, _ = discoveryClient.ServerGroupsAndResources()
```

Comment:
this has the potential to be an expensive call on a server with many API groups/versions... I wouldn't fall back to this just for a warning

Reply:
`kubectl` should cache these, but yes if the cache is out it could trigger an expensive call. I have removed the fallback as it does not seem necessary now.